cmake/modules/RootMacros.cmake

```diff
@@ -1731,15 +1731,22 @@ endfunction()
 
 #----------------------------------------------------------------------------
 # function ROOT_ADD_GTEST(<testsuite> source1 source2...
-#                        [WILLFAIL]
-#                        [COPY_TO_BUILDDIR file1 file2...] -- files to copy in the build directory
+#                        [WILLFAIL] Negate output of test
+#                        [COPY_TO_BUILDDIR file1 file2] Copy listed files when ctest invokes the test.
 #                        [LIBRARIES lib1 lib2...] -- Libraries to link against
 #                        [LABELS label1 label2...] -- Labels to annotate the test
 #                        [INCLUDE_DIRS label1 label2...] -- Extra target include directories
 #                        [REPEATS number] -- Repeats testsuite `number` times, stopping at the first failure.
-
+#                        [FAILREGEX ...] Fail test if this regex matches. Defaults to '(Fatal|Error|Warning) in <'
+# Creates a new googletest exectuable, and registers it as a test.
+# The test output will automatically be checked for ROOT's "Fatal/Error/Warning in <..."
```

Comment:
Wouldn't it make sense to move the default to EXPECTED_NO_DIAGS and keep the FAILREGEX always requiring input?

Reply:
I'm not sure I understand the question:
- If you don't pass anything, diags lead to test failures. Isn't that already what you are asking?
- Or are you asking for an extra argument `EXPECTED_NO_DIAGS` to be added? What should it do?

Reply:
@vgvassilev Maybe one could do the inverse:
- You suggest that every diag will fail a test (I agree, and this PR will introduce this feature).
- To allow diags in a test, you could set `EXPECT_DIAGS`.
- If you need a custom `FAILREGEX`, you can use `FAILREGEX "my.*Regex"`

Reply:
@hageboeck, I might have misread the code but we seemed to have `FAILREGEX ""` which I thought has a meaning to fail if there is any diagnostic output.

Reply:
Ah, no, it was actually the opposite:
- `""` == Failure on diagnostics
- `FAILREGEX ""` == No regex checking of output. Anything goes.
- `FAILREGEX "<regex>" == Search output for given regex.

I wanted to use some kind of flag to suppress the regex checking, but I didn't find a good name. There was `NO_FAILREGEX` in the beginning, but we removed it in favour of `FAILREGEX ""`. Maybe you can come up with a good name?
E.g. `DONT_CHECK_STDOUT`?